### PR TITLE
fix: ensure field directive is mapped to schema on relationship fields

### LINKED ISSUE #1772: Directives on field definitions are removed if there is a @oneToOne annotation
I'm not sure how general the problem is here, but in my generated schema for one of my fields with the following annotation:

```graphql
    """
    @oneToOne(key: 'company_id')
    """
```
the schema directives for that field in my model schema are removed in the generated schema.

### PR DESCRIPTION
**Pull request by craicoverflow**
Fixes #1772 

This fixes an issue where directives on relationship fields in the model were not added to the generated schema. because Graphback modifies the relationship fields to add extra metadata/filter arguments etc. 

Graphback was removing the old field and adding a new one with the same config + some modifications but we were losing the astNode from the field. I found that the better thing to do was to use [extendField](https://graphql-compose.github.io/docs/api/InterfaceTypeComposer.html#extendfield) from GraphQL Compose.

## Verification

I have added a test case in this PR. If you want to try it yourself, using the following model:

```graphql
directive @test on FIELD_DEFINITION
  """ @model """
type Note {
  id: ID!
  title: String!
  description: String
  """
  @oneToMany(field: 'note')
  """
  comments: [Comment]! @test
  """@oneToOne"""
  comment: Comment @test
}
""" @model """
type Comment {
  id: ID!
  text: String
  description: String
  note: Note! @test
}
```

### CODE REVIEW

File: packages/graphback-codegen-schema/tests/GraphQLSchemaCreatorTest.ts [hunk 1]
```diff
@@ -229,3 +229,50 @@ test('Creates CRUD resolvers for models', async () => {
...
+  // Note.comment field
+  expect(comment.astNode?.directives).toHaveLength(1)
+  expect(comment.astNode?.directives[0].name.value).toBe('test')
+
+  const commentType = schema.getType('Comment') as GraphQLObjectType
+  const { note } = commentType.getFields()
+
+  // Note.comments field
+  expect(note.astNode?.directives).toHaveLength(1)
+  expect(note.astNode?.directives[0].name.value).toBe('test')
```
Comment (machi1990) [inline packages/graphback-codegen-schema/tests/GraphQLSchemaCreatorTest.ts:277]: Thanks for adding these tests. 

Do you think we can have a snapshot test too?

**Comment by machi1990** [comment]
Could the fix be applied for this issue https://github.com/aerogear/graphback/issues/1767 or it is entirely has to do with grapql-compose?

File: packages/graphback-codegen-schema/tests/GraphQLSchemaCreatorTest.ts (hunk 1 above)
Comment (craicoverflow) [reply packages/graphback-codegen-schema/tests/GraphQLSchemaCreatorTest.ts]: Sure, pushed now :wink:

**Comment by craicoverflow** [comment]
> Could the fix be applied for this issue #1767 too or it is entirely has to do with grapql-compose?

This does not fix that issue unfortunately. I think that is entirely GraphQL Compose but I will do some investigating separate from this PR.

**Comment by wtrocki** [comment]
Ouch.. I merged wrong PR! Sorry @craicoverflow

**Comment by machi1990** [comment]
> > Could the fix be applied for this issue #1767 too or it is entirely has to do with grapql-compose?
> 
> This does not fix that issue unfortunately. I think that is entirely GraphQL Compose but I will do some investigating separate from this PR.

Okay I see. It was more a question for my own understanding. Thanks for answering.

**Pull request merged**